dbms/src/Dictionaries/DictionarySourceFactory.cpp

```diff
@@ -113,14 +122,18 @@ DictionarySourcePtr DictionarySourceFactory::create(
 		return std::make_unique<ClickHouseDictionarySource>(dict_struct, config, config_prefix + ".clickhouse",
 			sample_block, context);
 	}
+#if Poco_MongoDB_FOUND
```

Comment:
Need to add elseif statement, that will throw exception with meaningful exception message
(user must understand that dictionary source is known, but unavailable in current build).